### ISSUE TITLE
Configure SONiC OS on the Ethernet switch

### ISSUE THREAD

**Issue by shall1024** [OpenFabrics/fsdp_docs#29]
SONiC.master.125-dirty-20210216.022428 is now installed on the Edgecore Ethernet switch. Please let us know how you would like us to configure it, and we will apply the necessary configurations.

**Comment by dledford** [comment]
In the past, I did one vlan for no priority, which was intended not to be used with RoCE, made that the primary vlan for the RoCE ports, then created two other vlans accessible as tagged vlans on the RoCE ports that used PFC tagging and were intended to work properly with RoCE.  I found this setup to be problematic in terms of users not realizing they can't use the bare interface for RoCE work.  I'm going to suggest that we enable PFC on all ports, both RoCE and iWARP.  I would set priorities 3 and 5 as no drop (for now, that might change).  Then I would recommend three vlans for both RoCE and iWARP, both with a default access vlan and then each with two additional tagged vlans.  I don't think the numbers matter other than it would be nice if they didn't overlap with the IB P_Key numbers or the OPA P_Key numbers, that way the vlan tag/P_Key value can be used as the network number for the network, which enables some simplification of things.  Inside Red Hat, the IB P_Keys were from 0-11, OPA was 20-31, RoCE was vlans 40, 43, and 45, and iWARP was vlans 50-52.  Then the networks were 172.31.(vlan or P_Key number).(host specific number).  That allowed us to do some programmatic manipulation of hostnames and fabrics and know what our RDMA IP addresses were supposed to be.

**Comment by shall1024** [comment]
Will RoCE and iWARP be over different ports on the switch? If you intend to have an untagged VLAN for both RoCE and iWARP, they must be using different ports on the switch. Please let us know what port(s) we should use for each, or if any changes need to be made to the current planned VLAN configuration. Thank you.

**Comment by dledford** [comment]
Yes, RoCE and iWARP will be over different ports.  In the one case of the builder machine, we will configure it as a base RoCE device, but then add the iWARP ports as extra tagged vlans.  We have to configure its interface slightly differently as a result, but it works fine.  It allows the builder to serve dhcp without a problem.

**Comment by dledford** [comment]
VLAN setup is complete.  Finishing up the setup for the breakout ports.

**Comment by shall1024** [comment]
Dynamic Breakout Port commands are not working with SONiC, we have created an issue with the SONiC developers, still waiting on a response: https://github.com/Azure/sonic-buildimage/issues/7311

**Comment by shall1024** [comment]
We got a response from the SONiC developers on the issue, had issues pulling some packages required for the update from source so another issue was created [here](https://github.com/Azure/sonic-swss/issues/1734) that we're waiting on a response for.

**Comment by shall1024** [comment]
SONiC developers have suggested reinstalling a newer version of SONiC with the fix, currently working on this.

**Comment by shall1024** [comment]
The newer version of SONiC has a bug that I reported [here](https://github.com/Azure/sonic-buildimage/issues/7811). Interestingly enough, this version seems to have a completely different configuration file scheme than previous versions.

I attempted to install many older versions of SONiC that supported dynamic port breakout but they all had different problems, so for now, I think this is the best course of action is to wait for an answer on this issue.

**Comment by shall1024** [comment]
VLANs are confirmed to be working as of the week of 7/4/2021; still no update on the issue above.

**Comment by shall1024** [comment]
No new SONiC barefoot image has been generated since April, so I reinstalled an older version (#181) to the switch. Interfaces now actually appear with their current status on the switch.

The switch has been configured with the VLANs we want again, and they are, again, confirmed to be working.

**Comment by lylavoie** [comment]
Need to confirm if the switch hardware is supported by Cumulus, and if so, see if Nvidia might be willing to provide a license for use in the cluster.  This can be done with a free trial version of the NOS.

**Comment by shall1024** [comment]
The free trial we discussed [here](https://www.nvidia.com/en-us/networking/cumulus-linux-free-trial/) does not appear to be applicable to real, physical hardware, only to VMs/the cloud.

**Comment by dledford** [comment]
So, I just saw in one of the white papers that as of cumulous networks OS version 4.4, support for Broadcom chipsets has been removed.
